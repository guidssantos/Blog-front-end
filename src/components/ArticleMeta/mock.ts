import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-09-07T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'guilherme-dos-santos',
    displayName: 'Guilherme dos Santos',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'React',
      slug: 'react',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
