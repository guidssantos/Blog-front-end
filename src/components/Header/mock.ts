import { HeaderProps } from '.';

export default {
  blogName: 'Guilherme dos Santos',
  blogDescription: 'Uma descrição do Blog',
  logo: 'assets/images/guilherme.jfif',
  showText: true,
} as HeaderProps;
