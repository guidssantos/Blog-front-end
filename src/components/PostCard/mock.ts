import { data } from '../../components/api/dados.json';

export default data.posts[0];
