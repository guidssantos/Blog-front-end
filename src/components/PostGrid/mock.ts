import { data } from '../../components/api/dados.json';

export default { posts: data.posts };
