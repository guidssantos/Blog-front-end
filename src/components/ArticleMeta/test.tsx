import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ArticleMeta, ArticleMetaProps } from '.';

import mock from './mock';

const props: ArticleMetaProps = mock;

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(
      screen.getByRole('link', { name: 'Guilherme dos Santos' }),
    ).toHaveAttribute('href', '/author/guilherme-dos-santos');
    expect(screen.getByRole('link', { name: 'React' })).toHaveAttribute(
      'href',
      '/category/react',
    );
    expect(screen.getByRole('link', { name: 'JS' })).toHaveAttribute(
      'href',
      '/category/javascript',
    );
  });

  it('should format date', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByText('7 de set. de 2022')).toHaveAttribute(
      'datetime',
      props.createdAt,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with no author and categories', () => {
    const { container } = renderTheme(
      <ArticleMeta {...props} author={undefined} categories={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
