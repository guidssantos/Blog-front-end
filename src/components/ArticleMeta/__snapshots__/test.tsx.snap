// Vitest Snapshot v1

exports[`<ArticleMeta /> > should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
         Por 
      </span>
      <a
        href="/author/guilherme-dos-santos"
      >
        Guilherme dos Santos
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2022-09-07T15:16:09.246+00:00"
      >
        7 de set. de 2022
      </time>
      <span
        class="separator"
      >
        |
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/react"
          >
            React
          </a>
        </span>
        <span>
          <a
            href="/category/javascript"
          >
            JS
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`;
